weather_mv/loader_pipeline/pipeline.py

```diff
@@ -41,12 +56,80 @@ def pattern_to_uris(match_pattern: str) -> t.Iterable[str]:
         yield from [x.path for x in match.metadata_list]
 
 
+def _cleanup(bigquery_client: bigquery.Client, storage_client: storage.Client, canary_output_table: str,
+             canay_bucket_name: str, sig: t.Optional[t.Any] = None, frame: t.Optional[t.Any] = None) -> None:
+    bigquery_client.delete_table(canary_output_table, not_found_ok=True)
+    try:
+        storage_client.get_bucket(canay_bucket_name).delete(force=True)
+    except NotFound:
+        pass
+    if sig:
+        traceback.print_stack(frame)
+        sys.exit(0)
+
+
+def validate_region(output_table: str, temp_location: t.Optional[str] = None, region: t.Optional[str] = None) -> None:
+    """Validates non-compatible regions scenarios by performing sanity check."""
+    if not region and not temp_location:
+        raise ValueError('Invalid GCS location: None.')
+
+    bigquery_client = bigquery.Client()
+    storage_client = storage.Client()
+    canary_output_table = output_table + CANARY_OUTPUT_TABLE_SUFFIX + str(uuid.uuid4())
+    canay_bucket_name = CANARY_BUCKET_NAME + str(uuid.uuid4())
+
+    # Doing cleanup if operation get cut off midway.
+    # TODO : Should we handle some other signals ?
+    do_cleanup = partial(_cleanup, bigquery_client, storage_client, canary_output_table, canay_bucket_name)
+    signal.signal(signal.SIGINT, do_cleanup)
```

Comment:
I'm surprised that a `try..except. KeyboardInterrupt` couldn't handle this case: Have you tried that approach?

Reply:
Does this implementation have other advantages? For example, is this processed async?

Reply:
> When a user (not you, but others...) wants to interrupt a task he generally hits multiple times Ctrl-C. The first one will branch execution in the finally block. If another Ctrl-C occurs in the middle of the finally block because it contains slow operations like closing files, a new KeyboardInterrupt will be raised and nothing guarantees that the whole block will be executed.

SO : https://stackoverflow.com/questions/41280318/will-python-execute-finally-block-after-receiving-ctrlc

Reply:
Should we handle some other signals ? WDYT ? 
Currently handled SIGINT (Ctrl+c) and SIGTSTP (Ctrl+z)

Reply:
Thanks for the link to the SO post. TIL!

I think these signals are fine for now.